### ISSUE TITLE
Very high CPU usage

### ISSUE THREAD

**Issue by samheutmaker** [shanalikhan/code-settings-sync#1352]
This extension is causing the VSCode electron instance to use a lot of CPU. I have no more info.

**Comment by arbassic** [comment]
I noticed that too. It was also very hard to find out what extension caused about 50% CPU usage even when idle/no project opened. VSCode profiling through `code --status` shown nothing 😬

----

VS Code has its own settings synchronization for quite a while, so I think it is suggested to uninstall the extension for good.

**Comment by gabyx** [comment]
Yes, I ve seen the same here. Its the rigth extension which causes this.

**Comment by gabyx** [comment]
When you enable it will instantly jump to 50-100%

**Comment by DercilioFontes** [comment]
Same here.

Turning off and turning on!
![Screen Shot 2022-06-14 at 12 15 03 PM](https://user-images.githubusercontent.com/20172847/173627747-3118b5b4-000e-4e88-94bf-95fc024abfc1.png)

OS: macOS Monterey 12.4
VS Code: Version: 1.68.0

```text
Processes with high usage:

dercilio  9080 188.7  0.7 447914208 226240   ??  S    12:21pm   1:58.08 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node --inspect-port=0 /Applications/Visual Studio Code.app/Contents/Resources/app/out/bootstrap-fork --type=extensionHost --skipWorkspaceStorageLock
dercilio  9454   0.0  0.0 408637584   1792 s001  S+   12:40pm   0:00.00 grep --color=auto 9080
dercilio  9209   0.0  0.2 447073440  60704   ??  S    12:22pm   0:00.30 /Applications/Visual Studio Code.app/Contents/Frameworks/Code Helper.app/Contents/MacOS/Code Helper --ms-enable-electron-run-as-node /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=9080
```

**Comment by douira** [comment]
I can also confirm this, it only happens when this specific extension is enabled. Since VSCode already has this feature itself now, I've just uninstalled it. However, it would be nice if this bug could be fixed in this extension too.

**Comment by gnomeria** [comment]
I just did a vscode bisect, and pin down the incredible slowdown from my vscode is supposed to be this extension. Especially if I open multiple windows.

**Comment by yayzann** [comment]
Has anyone checked the processes/what they're communicating with?

**Comment by JIAZHEN** [comment]
+1 for high CPU usage. I have to disable this extension. Any fix?

**Comment by delongGao** [comment]
This has been an issue for me as well, have to disable it for now.

**Comment by sammcj** [comment]
It looks like there's a PR to fix it that needs to be merged - https://github.com/shanalikhan/code-settings-sync/pull/1379
